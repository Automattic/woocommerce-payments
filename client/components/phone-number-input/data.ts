// Do not edit this file directly.
// Generated by bin/build-metadata.js

/* eslint-disable */

const parse = ( data: any[] ) =>
	data.map( ( item: any[] ) => ( {
		alpha2: item[ 0 ] as string,
		code: item[ 1 ].toString() as string,
		priority: ( item[ 2 ] || 0 ) as number,
		start: item[ 3 ]?.map( String ) as string[] | undefined,
		lengths: item[ 4 ] as number[] | undefined,
	} ) );

const data = [
	[ 'AF', 93, , [ 7 ], [ 9 ] ],
	[ 'AL', 355, , [ 6 ], [ 9 ] ],
	[ 'DZ', 213, , [ 5, 6, 7 ], [ 9 ] ],
	[ 'AS', 1, 5, [ 684, 684733, 684258 ], [ 10 ] ],
	[ 'AD', 376, , [ 3, 4, 6 ], [ 6 ] ],
	[ 'AO', 244, , [ 9 ], [ 9 ] ],
	[ 'AI', 1, 6, [ 264, 2645, 2647 ], [ 10 ] ],
	[ 'AG', 1, 7, [ 268, 2687 ], [ 10 ] ],
	[ 'AR', 54, , [ 1, 2, 3 ], [ 8, 9, 10, 11, 12 ] ],
	[ 'AM', 374, , [ 3, 4, 5, 7, 9 ], [ 8 ] ],
	[ 'AW', 297, , [ 5, 6, 7, 9 ], [ 7 ] ],
	[ 'AC', 247, , , ],
	[ 'AU', 61, , [ 4 ], [ 9 ] ],
	[ 'AT', 43, , [ 6 ], [ 10, 11, 12, 13, 14 ] ],
	[ 'AZ', 994, , [ 4, 5, 6, 7 ], [ 9 ] ],
	[ 'BS', 1, 8, [ 242 ], [ 10 ] ],
	[ 'BH', 973, , [ 3 ], [ 8 ] ],
	[ 'BD', 880, , [ 1 ], [ 8, 9, 10 ] ],
	[ 'BB', 1, 9, [ 246 ], [ 10 ] ],
	[ 'BY', 375, , [ 25, 29, 33, 44 ], [ 9 ] ],
	[ 'BE', 32, , [ 4, 3 ], [ 9, 8 ] ],
	[ 'BZ', 501, , [ 6 ], [ 7 ] ],
	[ 'BJ', 229, , [ 4, 6, 9 ], [ 8 ] ],
	[ 'BM', 1, 10, [ 441, 4413, 4415, 4417 ], [ 10 ] ],
	[ 'BT', 975, , [ 17 ], [ 8 ] ],
	[ 'BO', 591, , [ 6, 7 ], [ 8 ] ],
	[ 'BA', 387, , [ 6 ], [ 8 ] ],
	[ 'BW', 267, , [ 71, 72, 73, 74, 75, 76 ], [ 8 ] ],
	[
		'BR',
		55,
		,
		[
			119,
			129,
			139,
			149,
			159,
			169,
			179,
			189,
			199,
			219,
			229,
			249,
			279,
			289,
			319,
			329,
			339,
			349,
			359,
			379,
			389,
			419,
			429,
			439,
			449,
			459,
			469,
			479,
			489,
			499,
			519,
			539,
			549,
			559,
			619,
			629,
			639,
			649,
			659,
			669,
			679,
			689,
			699,
			719,
			739,
			749,
			759,
			779,
			799,
			819,
			829,
			839,
			849,
			859,
			869,
			879,
			889,
			899,
			919,
			929,
			939,
			949,
			959,
			969,
			979,
			989,
			999,
		],
		[ 10, 11 ],
	],
	[ 'IO', 246, , , ],
	[ 'VG', 1, 11, [ 284 ], [ 10 ] ],
	[ 'BN', 673, , [ 7, 8 ], [ 7 ] ],
	[ 'BG', 359, , [ 87, 88, 89, 98, 99, 43 ], [ 8, 9 ] ],
	[ 'BF', 226, , [ 6, 7 ], [ 8 ] ],
	[ 'BI', 257, , [ 7, 29 ], [ 8 ] ],
	[ 'KH', 855, , [ 1, 6, 7, 8, 9 ], [ 8, 9 ] ],
	[ 'CM', 237, , [ 6 ], [ 9 ] ],
	[
		'CA',
		1,
		1,
		[
			204,
			226,
			236,
			249,
			250,
			289,
			306,
			343,
			365,
			367,
			368,
			387,
			403,
			416,
			418,
			431,
			437,
			438,
			450,
			474,
			506,
			514,
			519,
			548,
			579,
			581,
			584,
			587,
			604,
			613,
			639,
			647,
			672,
			705,
			709,
			742,
			778,
			780,
			782,
			807,
			819,
			825,
			867,
			873,
			902,
			905,
			263,
			354,
			468,
			600,
			683,
			753,
			428,
			382,
		],
		[ 10 ],
	],
	[ 'CV', 238, , [ 5, 9 ], [ 7 ] ],
	[ 'BQ', 599, 1, [ 3, 4, 7 ] ],
	[ 'KY', 1, 12, [ 345 ], [ 10 ] ],
	[ 'CF', 236, , [ 7 ], [ 8 ] ],
	[ 'TD', 235, , [ 6, 7, 9 ], [ 8 ] ],
	[ 'CL', 56, , [ 9 ], [ 9 ] ],
	[ 'CN', 86, , [ 13, 14, 15, 17, 18, 19, 16 ], [ 11 ] ],
	[ 'CX', 61, 2, [ 89164 ] ],
	[ 'CC', 61, 1, [ 89162 ] ],
	[ 'CO', 57, , [ 3 ], [ 10 ] ],
	[ 'KM', 269, , [ 3, 76 ], [ 7 ] ],
	[ 'CD', 243, , [ 8, 9 ], [ 9 ] ],
	[ 'CG', 242, , [ 0 ], [ 9 ] ],
	[ 'CK', 682, , [ 5, 7 ], [ 5 ] ],
	[ 'CR', 506, , [ 5, 6, 7, 8 ], [ 8 ] ],
	[ 'CI', 225, , [ 0, 4, 5, 6, 7, 8 ], [ 10 ] ],
	[ 'HR', 385, , [ 9 ], [ 8, 9 ] ],
	[ 'CU', 53, , [ 5 ], [ 8 ] ],
	[ 'CW', 599, , [ 5, 6 ], [ 7 ] ],
	[ 'CY', 357, , [ 9 ], [ 8 ] ],
	[ 'CZ', 420, , [ 6, 7 ], [ 9 ] ],
	[
		'DK',
		45,
		,
		[ 2, 30, 31, 40, 41, 42, 50, 51, 52, 53, 60, 61, 71, 81, 91, 92, 93 ],
		[ 8 ],
	],
	[ 'DJ', 253, , [ 77 ], [ 8 ] ],
	[ 'DM', 1, 13, [ 767 ], [ 10 ] ],
	[ 'DO', 1, 2, [ 809, 829, 849 ], [ 10 ] ],
	[ 'EC', 593, , [ 9 ], [ 9 ] ],
	[ 'EG', 20, , [ 1 ], [ 10, 8 ] ],
	[ 'SV', 503, , [ 7 ], [ 8 ] ],
	[ 'GQ', 240, , [ 222, 551 ], [ 9 ] ],
	[ 'ER', 291, , [ 1, 7, 8 ], [ 7 ] ],
	[ 'EE', 372, , [ 5, 81, 82, 83 ], [ 7, 8 ] ],
	[ 'SZ', 268, , , ],
	[ 'ET', 251, , [ 9 ], [ 9 ] ],
	[ 'FK', 500, , [ 5, 6 ], [ 5 ] ],
	[ 'FO', 298, , , [ 6 ] ],
	[ 'FJ', 679, , [ 2, 7, 8, 9 ], [ 7 ] ],
	[ 'FI', 358, , [ 4, 5 ], [ 9, 10 ] ],
	[ 'FR', 33, , [ 6, 7 ], [ 9 ] ],
	[ 'GF', 594, , [ 694 ], [ 9 ] ],
	[ 'PF', 689, , [ 8 ], [ 8 ] ],
	[ 'GA', 241, , [ 2, 3, 4, 5, 6, 7 ], [ 7 ] ],
	[ 'GM', 220, , [ 7, 9 ], [ 7 ] ],
	[ 'GE', 995, , [ 5, 7 ], [ 9 ] ],
	[ 'DE', 49, , [ 15, 16, 17 ], [ 10, 11 ] ],
	[ 'GH', 233, , [ 2, 5 ], [ 9 ] ],
	[ 'GI', 350, , [ 5 ], [ 8 ] ],
	[ 'GR', 30, , [ 6 ], [ 10 ] ],
	[ 'GL', 299, , [ 2, 4, 5 ], [ 6 ] ],
	[ 'GD', 1, 14, [ 473 ], [ 10 ] ],
	[ 'GP', 590, , [ 690 ], [ 9 ] ],
	[ 'GU', 1, 15, [ 671 ], [ 10 ] ],
	[ 'GT', 502, , [ 3, 4, 5 ], [ 8 ] ],
	[ 'GG', 44, 1, [ 1481, 7781, 7839, 7911 ] ],
	[ 'GN', 224, , [ 6 ], [ 9 ] ],
	[ 'GW', 245, , [ 5, 6, 7 ], [ 7 ] ],
	[ 'GY', 592, , [ 6 ], [ 7 ] ],
	[ 'HT', 509, , [ 3, 4 ], [ 8 ] ],
	[ 'HN', 504, , [ 3, 7, 8, 9 ], [ 8 ] ],
	[
		'HK',
		852,
		,
		[ 4, 5, 6, 70, 71, 72, 73, 81, 82, 83, 84, 85, 86, 87, 88, 89, 9 ],
		[ 8 ],
	],
	[ 'HU', 36, , [ 20, 30, 31, 50, 70 ], [ 9 ] ],
	[ 'IS', 354, , [ 6, 7, 8 ], [ 7 ] ],
	[ 'IN', 91, , [ 6, 7, 8, 9 ], [ 10 ] ],
	[ 'ID', 62, , [ 8 ], [ 9, 10, 11, 12 ] ],
	[ 'IR', 98, , [ 9 ], [ 10 ] ],
	[ 'IQ', 964, , [ 7 ], [ 10 ] ],
	[ 'IE', 353, , [ 82, 83, 84, 85, 86, 87, 88, 89 ], [ 9 ] ],
	[ 'IM', 44, 2, [ 1624, 74576, 7524, 7924, 7624 ] ],
	[ 'IL', 972, , [ 5 ], [ 9 ] ],
	[ 'IT', 39, , [ 3 ], [ 9, 10 ] ],
	[ 'JM', 1, 4, [ 876, 658 ], [ 10 ] ],
	[ 'JP', 81, , [ 70, 80, 90 ], [ 10 ] ],
	[ 'JE', 44, 3, [ 1534, 7509, 7700, 7797, 7829, 7937 ] ],
	[ 'JO', 962, , [ 7 ], [ 9 ] ],
	[ 'KZ', 7, 1, [ 33, 7, 70, 74, 77 ], [ 10 ] ],
	[ 'KE', 254, , [ 7, 1 ], [ 9 ] ],
	[ 'KI', 686, , [ 9, 30 ], [ 5 ] ],
	[ 'XK', 383, , , ],
	[ 'KW', 965, , [ 5, 6, 9 ], [ 8 ] ],
	[ 'KG', 996, , [ 5, 7, 8, 9 ], [ 9 ] ],
	[ 'LA', 856, , [ 20 ], [ 10 ] ],
	[ 'LV', 371, , [ 2 ], [ 8 ] ],
	[ 'LB', 961, , [ 3, 7, 8 ], [ 7, 8 ] ],
	[ 'LS', 266, , [ 5, 6 ], [ 8 ] ],
	[ 'LR', 231, , [ 4, 5, 6, 7 ], [ 7, 8 ] ],
	[ 'LY', 218, , [ 9 ], [ 9 ] ],
	[ 'LI', 423, , [ 7 ], [ 7 ] ],
	[ 'LT', 370, , [ 6 ], [ 8 ] ],
	[ 'LU', 352, , [ 6 ], [ 9 ] ],
	[ 'MO', 853, , [ 6 ], [ 8 ] ],
	[ 'MG', 261, , [ 3 ], [ 9 ] ],
	[ 'MW', 265, , [ 77, 88, 99 ], [ 9 ] ],
	[ 'MY', 60, , [ 1, 6 ], [ 9, 10, 8 ] ],
	[ 'MV', 960, , [ 7, 9 ], [ 7 ] ],
	[ 'ML', 223, , [ 6, 7 ], [ 8 ] ],
	[ 'MT', 356, , [ 7, 9 ], [ 8 ] ],
	[ 'MH', 692, , , [ 7 ] ],
	[ 'MQ', 596, , [ 696 ], [ 9 ] ],
	[ 'MR', 222, , , [ 8 ] ],
	[ 'MU', 230, , [ 5 ], [ 8 ] ],
	[ 'YT', 262, 1, [ 269, 639 ], [ 9 ] ],
	[ 'MX', 52, , [ '' ], [ 10, 11 ] ],
	[ 'FM', 691, , , [ 7 ] ],
	[ 'MD', 373, , [ 6, 7 ], [ 8 ] ],
	[ 'MC', 377, , [ 4, 6 ], [ 8, 9 ] ],
	[ 'MN', 976, , [ 5, 8, 9 ], [ 8 ] ],
	[ 'ME', 382, , [ 6 ], [ 8 ] ],
	[ 'MS', 1, 16, [ 664 ], [ 10 ] ],
	[ 'MA', 212, , [ 6, 7 ], [ 9 ] ],
	[ 'MZ', 258, , [ 8 ], [ 9 ] ],
	[ 'MM', 95, , [ 9 ], [ 8, 9, 10 ] ],
	[ 'NA', 264, , [ 60, 81, 82, 85 ], [ 9 ] ],
	[ 'NR', 674, , [ 555 ], [ 7 ] ],
	[ 'NP', 977, , [ 97, 98 ], [ 10 ] ],
	[ 'NL', 31, , [ 6 ], [ 9 ] ],
	[ 'NC', 687, , [ 7, 8, 9 ], [ 6 ] ],
	[ 'NZ', 64, , [ 2 ], [ 8, 9, 10 ] ],
	[ 'NI', 505, , [ 8 ], [ 8 ] ],
	[ 'NE', 227, , [ 9 ], [ 8 ] ],
	[ 'NG', 234, , [ 70, 80, 81, 90, 91 ], [ 10 ] ],
	[ 'NU', 683, , , [ 4 ] ],
	[ 'NF', 672, , [ 5, 8 ], [ 5 ] ],
	[ 'KP', 850, , , ],
	[ 'MK', 389, , [ 7 ], [ 8 ] ],
	[ 'MP', 1, 17, [ 670 ], [ 10 ] ],
	[ 'NO', 47, , [ 4, 9 ], [ 8 ] ],
	[ 'OM', 968, , [ 9 ], [ 8 ] ],
	[ 'PK', 92, , [ 3 ], [ 10 ] ],
	[ 'PW', 680, , , [ 7 ] ],
	[ 'PS', 970, , [ 5 ], [ 9 ] ],
	[ 'PA', 507, , [ 6 ], [ 8 ] ],
	[ 'PG', 675, , [ 7 ], [ 8 ] ],
	[ 'PY', 595, , [ 9 ], [ 9 ] ],
	[ 'PE', 51, , [ 9 ], [ 9 ] ],
	[ 'PH', 63, , [ 9 ], [ 10 ] ],
	[ 'PL', 48, , [ 4, 5, 6, 7, 8 ], [ 9 ] ],
	[ 'PT', 351, , [ 9 ], [ 9 ] ],
	[ 'PR', 1, 3, [ 787, 939 ], [ 10 ] ],
	[ 'QA', 974, , [ 3, 5, 6, 7 ], [ 8 ] ],
	[ 'RE', 262, , [ 692, 693 ], [ 9 ] ],
	[ 'RO', 40, , [ 7 ], [ 9 ] ],
	[ 'RU', 7, , [ 9, 495, 498, 499, 835 ], [ 10 ] ],
	[ 'RW', 250, , [ 7 ], [ 9 ] ],
	[ 'BL', 590, 1, , ],
	[ 'SH', 290, , , [ 4 ] ],
	[ 'KN', 1, 18, [ 869 ], [ 10 ] ],
	[ 'LC', 1, 19, [ 758 ], [ 10 ] ],
	[ 'MF', 590, 2, , ],
	[ 'PM', 508, , [ 55, 41 ], [ 6 ] ],
	[ 'VC', 1, 20, [ 784 ], [ 10 ] ],
	[ 'WS', 685, , [ 7 ], [ 7 ] ],
	[ 'SM', 378, , [ 3, 6 ], [ 10 ] ],
	[ 'ST', 239, , [ 98, 99 ], [ 7 ] ],
	[ 'SA', 966, , [ 5 ], [ 9 ] ],
	[ 'SN', 221, , [ 7 ], [ 9 ] ],
	[ 'RS', 381, , [ 6 ], [ 8, 9 ] ],
	[ 'SC', 248, , [ 2 ], [ 7 ] ],
	[
		'SL',
		232,
		,
		[ 21, 25, 30, 33, 34, 40, 44, 50, 55, 76, 77, 78, 79, 88 ],
		[ 8 ],
	],
	[ 'SG', 65, , [ 8, 9 ], [ 8 ] ],
	[ 'SX', 1, 21, [ 721 ], [ 10 ] ],
	[ 'SK', 421, , [ 9 ], [ 9 ] ],
	[ 'SI', 386, , [ 3, 4, 5, 6, 7 ], [ 8 ] ],
	[ 'SB', 677, , [ 7, 8 ], [ 7 ] ],
	[ 'SO', 252, , [ 61, 62, 63, 65, 66, 68, 69, 71, 90 ], [ 9 ] ],
	[ 'ZA', 27, , [ 1, 2, 3, 4, 5, 6, 7, 8 ], [ 9 ] ],
	[ 'KR', 82, , [ 1 ], [ 9, 10 ] ],
	[ 'SS', 211, , [ 9 ], [ 9 ] ],
	[ 'ES', 34, , [ 6, 7 ], [ 9 ] ],
	[ 'LK', 94, , [ 7 ], [ 9 ] ],
	[ 'SD', 249, , [ 9 ], [ 9 ] ],
	[ 'SR', 597, , [ 6, 7, 8 ], [ 7 ] ],
	[ 'SJ', 47, 1, [ 79 ], [ 8 ] ],
	[ 'SE', 46, , [ 7 ], [ 9 ] ],
	[ 'CH', 41, , [ 74, 75, 76, 77, 78, 79 ], [ 9 ] ],
	[ 'SY', 963, , [ 9 ], [ 9 ] ],
	[ 'TW', 886, , [ 9 ], [ 9 ] ],
	[ 'TJ', 992, , [ 9 ], [ 9 ] ],
	[ 'TZ', 255, , [ 7, 6 ], [ 9 ] ],
	[ 'TH', 66, , [ 6, 8, 9 ], [ 9 ] ],
	[ 'TL', 670, , [ 7 ], [ 8 ] ],
	[ 'TG', 228, , [ 9 ], [ 8 ] ],
	[ 'TK', 690, , , [ 4 ] ],
	[ 'TO', 676, , , [ 5 ] ],
	[ 'TT', 1, 22, [ 868 ], [ 10 ] ],
	[ 'TN', 216, , [ 2, 4, 5, 9 ], [ 8 ] ],
	[ 'TR', 90, , [ 5 ], [ 10 ] ],
	[ 'TM', 993, , [ 6 ], [ 8 ] ],
	[ 'TC', 1, 23, [ 649, 6492, 6493, 6494 ], [ 10 ] ],
	[ 'TV', 688, , , [ 5 ] ],
	[ 'VI', 1, 24, [ 340 ], [ 10 ] ],
	[ 'UG', 256, , [ 7 ], [ 9 ] ],
	[ 'UA', 380, , [ 39, 50, 63, 66, 67, 68, 73, 9 ], [ 9 ] ],
	[ 'AE', 971, , [ 5 ], [ 9 ] ],
	[ 'GB', 44, , [ 7 ], [ 10 ] ],
	[
		'US',
		1,
		,
		[
			201,
			202,
			203,
			205,
			206,
			207,
			208,
			209,
			210,
			212,
			213,
			214,
			215,
			216,
			217,
			218,
			219,
			220,
			223,
			224,
			225,
			227,
			228,
			229,
			231,
			234,
			239,
			240,
			248,
			251,
			252,
			253,
			254,
			256,
			260,
			262,
			267,
			269,
			270,
			272,
			274,
			276,
			278,
			281,
			283,
			301,
			302,
			303,
			304,
			305,
			307,
			308,
			309,
			310,
			312,
			313,
			314,
			315,
			316,
			317,
			318,
			319,
			320,
			321,
			323,
			325,
			327,
			330,
			331,
			332,
			334,
			336,
			337,
			339,
			341,
			346,
			347,
			351,
			352,
			360,
			361,
			364,
			369,
			380,
			385,
			386,
			401,
			402,
			404,
			405,
			406,
			407,
			408,
			409,
			410,
			412,
			413,
			414,
			415,
			417,
			419,
			423,
			424,
			425,
			430,
			432,
			434,
			435,
			440,
			441,
			442,
			443,
			445,
			447,
			458,
			463,
			464,
			469,
			470,
			475,
			478,
			479,
			480,
			484,
			501,
			502,
			503,
			504,
			505,
			507,
			508,
			509,
			510,
			512,
			513,
			515,
			516,
			517,
			518,
			520,
			530,
			531,
			534,
			539,
			540,
			541,
			551,
			557,
			559,
			561,
			562,
			563,
			564,
			567,
			570,
			571,
			572,
			573,
			574,
			575,
			580,
			582,
			585,
			586,
			601,
			602,
			603,
			605,
			606,
			607,
			608,
			609,
			610,
			612,
			614,
			615,
			616,
			617,
			618,
			619,
			620,
			623,
			626,
			627,
			628,
			629,
			630,
			631,
			636,
			640,
			641,
			646,
			650,
			651,
			656,
			657,
			659,
			660,
			661,
			662,
			667,
			669,
			678,
			679,
			680,
			681,
			682,
			689,
			701,
			702,
			703,
			704,
			706,
			707,
			708,
			712,
			713,
			714,
			715,
			716,
			717,
			718,
			719,
			720,
			724,
			725,
			726,
			727,
			730,
			731,
			732,
			734,
			737,
			740,
			743,
			747,
			752,
			754,
			757,
			760,
			762,
			763,
			764,
			765,
			769,
			770,
			771,
			772,
			773,
			774,
			775,
			779,
			781,
			785,
			786,
			787,
			801,
			802,
			803,
			804,
			805,
			806,
			808,
			810,
			812,
			813,
			814,
			815,
			816,
			817,
			818,
			820,
			828,
			830,
			831,
			832,
			835,
			838,
			840,
			843,
			845,
			847,
			848,
			850,
			854,
			856,
			857,
			858,
			859,
			860,
			862,
			863,
			864,
			865,
			870,
			872,
			878,
			901,
			903,
			904,
			906,
			907,
			908,
			909,
			910,
			912,
			913,
			914,
			915,
			916,
			917,
			918,
			919,
			920,
			925,
			927,
			928,
			929,
			930,
			931,
			934,
			935,
			936,
			937,
			938,
			939,
			940,
			941,
			945,
			947,
			949,
			951,
			952,
			954,
			956,
			957,
			959,
			970,
			971,
			972,
			973,
			975,
			978,
			979,
			980,
			984,
			985,
			986,
			989,
			888,
			800,
			833,
			844,
			855,
			866,
			877,
			279,
			340,
			983,
			448,
			943,
			363,
			326,
			839,
			826,
			948,
		],
		[ 10 ],
	],
	[ 'UY', 598, , [ 9 ], [ 8 ] ],
	[ 'UZ', 998, , [ 9, 88, 33 ], [ 9 ] ],
	[ 'VU', 678, , [ 5, 7 ], [ 7 ] ],
	[ 'VA', 39, 1, [ '06698' ] ],
	[ 'VE', 58, , [ 4 ], [ 10 ] ],
	[ 'VN', 84, , [ 8, 9, 3, 7, 5 ], [ 9 ] ],
	[ 'WF', 681, , , [ 6 ] ],
	[ 'EH', 212, 1, [ 5288, 5289 ] ],
	[ 'YE', 967, , [ 7 ], [ 9 ] ],
	[ 'ZM', 260, , [ 9, 7 ], [ 9 ] ],
	[ 'ZW', 263, , [ 71, 73, 77, 78 ], [ 9 ] ],
	[ 'AX', 358, 1, [ 18 ], [ 6, 7, 8 ] ],
];

export default parse( data );
export type DataType = ReturnType< typeof parse >;
