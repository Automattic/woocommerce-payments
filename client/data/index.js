/** @format */

/**
 * Internal dependencies
 */
import { STORE_NAME } from './constants';
import { initStore } from './store';

initStore();

export const WCPAY_STORE_NAME = STORE_NAME;

// We only ask for hooks when importing directly from 'data'.
export * from './deposits/hooks';
export * from './transactions/hooks';
export * from './charges/hooks';
export * from './timeline/hooks';
export * from './disputes/hooks';
export * from './settings/hooks';
export * from './multi-currency';
