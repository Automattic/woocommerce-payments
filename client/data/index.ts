/** @format */

/**
 * Internal dependencies
 */
import { STORE_NAME } from './constants';
import { initStore } from './store';

initStore();

// eslint-disable-next-line @typescript-eslint/naming-convention
export const WCPAY_STORE_NAME = STORE_NAME;

// We only ask for hooks when importing directly from 'wcpay/data'.
export * from './deposits/hooks';
export * from './transactions/hooks';
export * from './charges/hooks';
export * from './timeline/hooks';
export * from './disputes/hooks';
export * from './settings/hooks';
export * from './multi-currency';
export * from './card-readers/hooks';
export * from './capital/hooks';
export * from './documents/hooks';
export * from './payment-intents/hooks';
export * from './authorizations/hooks';
