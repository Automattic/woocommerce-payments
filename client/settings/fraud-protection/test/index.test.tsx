/**
 * External dependencies
 */
import React from 'react';
import { render } from '@testing-library/react';
import { useDispatch } from '@wordpress/data';

/**
 * Internal dependencies
 */
import FraudProtection from '..';
import {
	useCurrentProtectionLevel,
	useAdvancedFraudProtectionSettings,
	useSettings,
} from 'wcpay/data';
import WCPaySettingsContext from '../../wcpay-settings-context';

declare const global: {
	wcpaySettings: {
		fraudProtection: {
			isWelcomeTourDismissed: boolean;
		};
	};
};

jest.mock( 'wcpay/data', () => ( {
	useAdvancedFraudProtectionSettings: jest.fn(),
	useCurrentProtectionLevel: jest.fn(),
	useSettings: jest.fn(),
} ) );

jest.mock( '@wordpress/data', () => ( {
	useDispatch: jest.fn(),
} ) );

jest.mock( '@woocommerce/components', () => ( {
	TourKit: () => <div>Tour Component</div>,
} ) );

const mockUseCurrentProtectionLevel = useCurrentProtectionLevel as jest.MockedFunction<
	() => [ string, ( level: string ) => void ]
>;

const mockUseAdvancedFraudProtectionSettings = useAdvancedFraudProtectionSettings as jest.MockedFunction<
	() => [ any[] | string, ( settings: string ) => void ]
>;

const mockUseSettings = useSettings as jest.MockedFunction<
	() => {
		settings: any;
		isLoading: boolean;
		saveSettings: () => void;
		isSaving: boolean;
	}
>;

const mockUseDispatch = useDispatch as jest.MockedFunction< any >;

describe( 'FraudProtection', () => {
	beforeEach( () => {
		mockUseCurrentProtectionLevel.mockReturnValue( [
			'standard',
			jest.fn(),
		] );

		mockUseAdvancedFraudProtectionSettings.mockReturnValue( [
			[],
			jest.fn(),
		] );
		mockUseSettings.mockReturnValue( {
			settings: {},
			isSaving: false,
			saveSettings: jest.fn(),
			isLoading: false,
		} );
		mockUseDispatch.mockReturnValue( { updateOptions: jest.fn() } );

		global.wcpaySettings = {
			fraudProtection: {
				isWelcomeTourDismissed: false,
			},
		};
	} );

	it( 'should render correctly', () => {
		Object.defineProperty( window, 'location', {
			configurable: true,
			enumerable: true,
			value: {
				search:
					'?page=wc-settings&tab=checkout&anchor=%23fp-settings&section=woocommerce_payments/',
			},
		} );

		const { container: fraudProtectionSettings } = render(
			<WCPaySettingsContext.Provider
				value={ { isWelcomeTourDismissed: true } as any }
			>
				<FraudProtection />
			</WCPaySettingsContext.Provider>
		);

		expect( fraudProtectionSettings ).toMatchSnapshot();
	} );

	it( 'should render with the welcome tour dismissed', () => {
		global.wcpaySettings.fraudProtection.isWelcomeTourDismissed = true;

		const { container: fraudProtectionSettings } = render(
			<WCPaySettingsContext.Provider
				value={ { isWelcomeTourDismissed: true } as any }
			>
				<FraudProtection />
			</WCPaySettingsContext.Provider>
		);

		expect( fraudProtectionSettings ).toMatchSnapshot();
	} );
} );
