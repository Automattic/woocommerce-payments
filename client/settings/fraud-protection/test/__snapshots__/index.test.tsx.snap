// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FraudProtection should render correctly 1`] = `
<div>
  <div
    class="components-card is-size-medium fraud-protection css-1xs3c37-CardUI e1q7k77g0"
  >
    <div
      class="components-card__body is-size-medium wcpay-card-body css-xmjzce-BodyUI e1q7k77g3"
    >
      <h4>
        <span
          id="fraud-protection-card-title"
        >
          Set your payment risk level
        </span>
      </h4>
      <fieldset>
        <ul>
          <li>
            <div
              class="fraud-protection-radio-wrapper"
            >
              <input
                class="fraud-protection-radio-wrapper__item"
                id="fraud-protection__basic-level"
                name="fraud-protection-level-select"
                type="radio"
                value="basic"
              />
              <label
                class="fraud-protection-radio-wrapper__item"
                for="fraud-protection__basic-level"
              >
                Basic
              </label>
              <svg
                class="gridicon gridicons-help-outline fraud-protection__help-icon needs-offset"
                height="18"
                title="Basic level help icon"
                viewBox="0 0 24 24"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M12 4c4.411 0 8 3.589 8 8s-3.589 8-8 8-8-3.589-8-8 3.589-8 8-8m0-2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm4 8a4 4 0 00-8 0h2c0-1.103.897-2 2-2s2 .897 2 2-.897 2-2 2a1 1 0 00-1 1v2h2v-1.141A3.991 3.991 0 0016 10zm-3 6h-2v2h2v-2z"
                  />
                </g>
              </svg>
            </div>
            <p
              class="fraud-protection__text--help-text"
            >
              Provides the base level of platform protection.
            </p>
          </li>
          <hr
            class="fraud-protection__list-divider"
          />
          <li
            class="fraud-protection__advanced-level-container"
          >
            <label
              for="fraud-protection-level-select_advanced-level"
            >
              <div
                class="fraud-protection-radio-wrapper"
              >
                <input
                  id="fraud-protection-level-select_advanced-level"
                  name="fraud-protection-level-select"
                  type="radio"
                  value="advanced"
                />
                <p
                  class="fraud-protection-radio-wrapper__item"
                >
                  Advanced
                </p>
              </div>
              <p
                class="fraud-protection__text--help-text"
              >
                Allows you to fine-tune the level of filtering according to your business needs.
              </p>
            </label>
            <button
              class="components-button is-secondary"
              disabled=""
              type="button"
            >
              Configure
            </button>
          </li>
        </ul>
      </fieldset>
    </div>
  </div>
  <div
    id="fraud-protection-welcome-tour-first-step"
  />
  <div>
    Tour Component
  </div>
</div>
`;

exports[`FraudProtection should render with the welcome tour dismissed 1`] = `
<div>
  <div
    class="components-card is-size-medium fraud-protection css-1xs3c37-CardUI e1q7k77g0"
  >
    <div
      class="components-card__body is-size-medium wcpay-card-body css-xmjzce-BodyUI e1q7k77g3"
    >
      <h4>
        <span
          id="fraud-protection-card-title"
        >
          Set your payment risk level
        </span>
      </h4>
      <fieldset>
        <ul>
          <li>
            <div
              class="fraud-protection-radio-wrapper"
            >
              <input
                class="fraud-protection-radio-wrapper__item"
                id="fraud-protection__basic-level"
                name="fraud-protection-level-select"
                type="radio"
                value="basic"
              />
              <label
                class="fraud-protection-radio-wrapper__item"
                for="fraud-protection__basic-level"
              >
                Basic
              </label>
              <svg
                class="gridicon gridicons-help-outline fraud-protection__help-icon needs-offset"
                height="18"
                title="Basic level help icon"
                viewBox="0 0 24 24"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M12 4c4.411 0 8 3.589 8 8s-3.589 8-8 8-8-3.589-8-8 3.589-8 8-8m0-2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2zm4 8a4 4 0 00-8 0h2c0-1.103.897-2 2-2s2 .897 2 2-.897 2-2 2a1 1 0 00-1 1v2h2v-1.141A3.991 3.991 0 0016 10zm-3 6h-2v2h2v-2z"
                  />
                </g>
              </svg>
            </div>
            <p
              class="fraud-protection__text--help-text"
            >
              Provides the base level of platform protection.
            </p>
          </li>
          <hr
            class="fraud-protection__list-divider"
          />
          <li
            class="fraud-protection__advanced-level-container"
          >
            <label
              for="fraud-protection-level-select_advanced-level"
            >
              <div
                class="fraud-protection-radio-wrapper"
              >
                <input
                  id="fraud-protection-level-select_advanced-level"
                  name="fraud-protection-level-select"
                  type="radio"
                  value="advanced"
                />
                <p
                  class="fraud-protection-radio-wrapper__item"
                >
                  Advanced
                </p>
              </div>
              <p
                class="fraud-protection__text--help-text"
              >
                Allows you to fine-tune the level of filtering according to your business needs.
              </p>
            </label>
            <button
              class="components-button is-secondary"
              disabled=""
              type="button"
            >
              Configure
            </button>
          </li>
        </ul>
      </fieldset>
    </div>
  </div>
</div>
`;
